### PR TITLE
fix(test): use fixed seed to improve reproducibility.

### LINKED ISSUE #2550: Flaky test in HNSW index unit test
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Version

unstable

### Minimal reproduce step

See https://github.com/apache/kvrocks/actions/runs/10971334789/job/30466698610?pr=2538#step:15:173

### What did you expect to see?

CI passed

### What did you see instead?

CI failure

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by LindaSummer**
# Issue

Fix: #2550 

Unit test's random generation may make result unstable to expected result.

# Proposed Changes

Use fixed seed for hnsw index unit test for improving reproducibility.

### CODE REVIEW

File: src/search/hnsw_indexer.h [hunk 1]
```diff
@@ -92,7 +92,10 @@ struct HnswIndex {
   std::mt19937 generator;
   double m_level_normalization_factor;
 
-  HnswIndex(const SearchKey& search_key, HnswVectorFieldMetadata* vector, engine::Storage* storage);
+  HnswIndex(const SearchKey& search_key, HnswVectorFieldMetadata* vector, engine::Storage* storage,
+            std::random_device::result_type seed);
+  HnswIndex(const SearchKey& search_key, HnswVectorFieldMetadata* vector, engine::Storage* storage)
+      : HnswIndex(search_key, vector, storage, std::random_device()()) {}
```
Comment (Beihao-Zhou) [inline src/search/hnsw_indexer.h]: Maybe we can use a single constructor with a default argument? 
```
HnswIndex(const SearchKey& search_key, 
          HnswVectorFieldMetadata* vector, 
          engine::Storage* storage, 
          std::random_device::result_type seed = std::random_device()())
{
    ...
}
```

**Comment by Beihao-Zhou** [comment]
The rest LGTM :)

**Comment by Beihao-Zhou** [comment]
The rest LGTM :)

File: src/search/hnsw_indexer.h (hunk 1 above)
Comment (LindaSummer) [reply src/search/hnsw_indexer.h]: Hi @Beihao-Zhou ,

Thanks for your review suggestions!
I have updated related constructors. 😊

Best Regards,
Edward

**Review by git-hulk — APPROVED** [review]

**Review by PragmaTwice — APPROVED** [review]

**Review by Beihao-Zhou — APPROVED** [review]

**Pull request merged**